index.html

```diff
@@ -0,0 +1,475 @@
+
+## Example: Let's go to the moon
+
+*This section is non-normative.*
+
+We'll refer to this example throughout the document. The example consists of two example schemas—a `rockets` schema that holds data about rockets, and an `astronauts` schema that holds information about the people flying on them—along with a CSDL that composes them.
+
+```graphql
+#::[Rockets example service]
+#::a.not-csdl
+type Query {
+  rockets: [Rocket]!
+}
+
+type Rocket @key(fields: "id") {
+  id: String! 
+  name: String!
+  captain: Astronaut @provides(fields: "tripId")
+}
+
+extend type Astronaut @key(fields: "id") {
+  id: String! @external
+  tripId: String! @external
+  rocket: Rocket!
+}
+```
+
+```graphql
+#::[Astronauts example service]
+#::a.not-csdl
+type Query {
+  astronauts: [Astronaut]!
+}
+
+type Astronaut @key(fields: "id") {
+  id: String! 
+  name: String!
+  tripId: String!
+}
+
+extend type Rocket @key(fields: "id") {
+  id: String! @external
+  astronaut: Astronaut!
+}
+```
+
+This example is non-normative, and represents just one possible way to compose [rockets](#rockets) and [astronauts](#astronauts) into CSDL:
+
+```graphql
+#::[CSDL composing `rockets` and `astronauts`]
+schema @using(spec: "https://specs.apollo.dev/csdl/v1") {
+  query: Query
+}
+
+enum csdl_Graph {
+  ROCKETS    @csdl_endpoint(url: "https://rockets.example.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.example.com"),
+}
+
+type Query {
+  rockets: [Rocket]!       @csdl_resolve(graph: ROCKETS)
+  astronauts: [Astronaut]! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Astronaut
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!     @csdl_resolve(graph: ASTRONAUTS)
+  name: String!   @csdl_resolve(graph: ASTRONAUTS)
+  tripId: String! @csdl_resolve(graph: ASTRONAUTS)
+  rocket: Rocket! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Rocket 
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!           @csdl_resolve(graph: ROCKETS)
+  name: String!         @csdl_resolve(graph: ROCKETS)
+  astronaut: Astronaut! @csdl_resolve(graph: ASTRONAUTS)
+  captain: Astronaut    @csdl_resolve(graph: ASTRONAUTS,
+                                      provides: "{ tripId }")
+}
+```
+
+The meaning of these directives is explored the [Directives](#directives) section.
+
+## Actors
+
+```mermaid
+#::[Actors and roles within an example composition pipeline]
+flowchart TB
+    subgraph A [subgraph A]
+      schemaA([schema A])
+      endpointA([endpoint A])
+    end    
+    subgraph B [subgraph B]
+      schemaB([schema B])
+      endpointB([endpoint B])
+    end
+    subgraph C [subgraph C]
+      schemaC([schema C])
+      endpointC([endpoint C])
+    end
+    subgraph Producer
```

Comment:
If you can get the words Producer and Consumer to render somewhere that isn't the middle of an arrow from above I'd find that clearer — it wasn't immediately obvious to me that those words were labeling the dashed box rather than the arrow.

Reply:
oh believe me i tried. might hammer on it more later, but it's surprisingly tricky to move these things around in mermaid